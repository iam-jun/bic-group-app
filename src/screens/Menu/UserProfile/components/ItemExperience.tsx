import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React from 'react';
import { StyleSheet, View } from 'react-native';
import Text from '~/beinComponents/Text';
import { useBaseHook } from '~/hooks';
import { IUserWorkExperience } from '~/interfaces/IAuth';
import { spacing } from '~/theme';
import { formatDate } from '~/utils/formatData';
import InfoItem from './InfoItem';

const ItemExperience = ({
  company,
  titlePosition,
  startDate,
  currentlyWorkHere,
  endDate,
  description,
  location,
}: IUserWorkExperience) => {
  const { t } = useBaseHook();
  const theme: ExtendedTheme = useTheme();
  const styles = themeStyles(theme);

  return (
    <View style={styles.container}>
      <View style={styles.titleCompany}>
        <View style={styles.titleLine} />
        <Text.SubtitleL color={theme.colors.neutral20}>
          {`${t('settings:text_work')} ${t('common:text_at')} `}
          <Text.SubtitleL>{company}</Text.SubtitleL>
        </Text.SubtitleL>
      </View>
      <InfoItem
        title="settings:text_title_position"
        value={titlePosition}
      />
      <InfoItem
        title="settings:text_title_division"
        value={location}
      />
      <InfoItem
        title="common:text_start_date"
        value={formatDate(
          startDate, 'MMM D, YYYY',
        )}
      />
      <InfoItem
        title="common:text_end_date"
        value={currentlyWorkHere
          ? `${t('common:text_to')} ${t('common:text_present')}`
          : endDate
            ? `${t('common:text_to')} ${formatDate(
              endDate,
              'MMM D, YYYY',
            )}`
            : ''}
      />
    </View>
  )
}

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create({
    container: {
      paddingBottom: spacing.padding.extraLarge,
    },
    titleCompany: {
      flexDirection: 'row',
      paddingBottom: spacing.padding.large,
    },
    titleLine: {
      width: 1,
      height: 22,
      backgroundColor: colors.neutral20,
      marginRight: spacing.padding.base,

    },
    infoItem: {
      paddingBottom: spacing.padding.large,
    },
  })
}

export default ItemExperience;
