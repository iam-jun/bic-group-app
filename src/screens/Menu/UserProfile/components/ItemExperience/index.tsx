import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React, { FC } from 'react';
import { StyleSheet, View } from 'react-native';
import { useDispatch } from 'react-redux';
import Text from '~/beinComponents/Text';
import { useBaseHook } from '~/hooks';
import { IUserWorkExperience } from '~/interfaces/IAuth';
import { spacing } from '~/theme';
import { formatDate } from '~/utils/formatData';
import { getEndDateText } from '../../helper';
import InfoItem from '../InfoItem';
import EditButton from '../../components/EditButton';
import ViewSpacing from '~/beinComponents/ViewSpacing';
import DeleteButton from '../DeleteButton';
import menuActions from '~/storeRedux/menu/actions';
import { useRootNavigation } from '~/hooks/navigation';
import mainStack from '~/router/navigator/MainStack/stack';

type ItemExperienceProps = {
  item: IUserWorkExperience;
  isCurrentUser: boolean;
};

const ItemExperience: FC<ItemExperienceProps> = ({
  item,
  isCurrentUser,
}) => {
  const {
    id,
    company,
    titlePosition,
    startDate,
    currentlyWorkHere,
    endDate,
    location,
    description,
  } = item;
  const dispatch = useDispatch();
  const { t } = useBaseHook();
  const theme: ExtendedTheme = useTheme();
  const styles = themeStyles(theme);
  const { rootNavigation } = useRootNavigation();

  const editItemExperience = () => {
    dispatch(menuActions.setSelectedWorkItem(item));
    rootNavigation.navigate(mainStack.addWork);
  };

  const deleteItemExperience = () => {
    dispatch(menuActions.deleteWorkExperience(
      id,
    ));
  };

  return (
    <View testID="item_experience">
      <View style={styles.row}>
        <View style={styles.titleCompany}>
          <View style={styles.titleLine} />
          <Text.SubtitleL color={theme.colors.neutral20}>
            {`${t('settings:text_work')} ${t('common:text_at')} `}
            <Text.SubtitleL>{company}</Text.SubtitleL>
          </Text.SubtitleL>
        </View>
        <View style={styles.rowAction}>
          <DeleteButton
            isCurrentUser={isCurrentUser}
            onPress={deleteItemExperience}
            icon="TrashCanSolid"
          />
          <ViewSpacing width={spacing.margin.small} />
          <EditButton
            isCurrentUser={isCurrentUser}
            onPress={editItemExperience}
            icon="PenToSquareSolid"
          />
        </View>
      </View>
      <InfoItem title="settings:text_title_position" value={titlePosition} />
      <InfoItem title="settings:title_location" value={location} />
      <InfoItem
        title="common:text_start_date"
        value={formatDate(startDate, 'MMM D, YYYY')}
      />
      <InfoItem
        title="common:text_end_date"
        value={getEndDateText(t, currentlyWorkHere, endDate)}
      />
      <InfoItem
        title="common:text_description"
        value={description}
      />
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create({
    titleCompany: {
      flexDirection: 'row',
      flex: 1,
      marginRight: spacing.margin.small,
    },
    titleLine: {
      width: 1,
      height: 22,
      backgroundColor: colors.neutral20,
      marginRight: spacing.padding.base,
    },
    infoItem: {
      paddingBottom: spacing.padding.large,
    },
    row: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      paddingBottom: spacing.padding.large,
    },
    rowAction: {
      marginLeft: spacing.margin.base,
      flexDirection: 'row',
    },
  });
};

export default ItemExperience;
