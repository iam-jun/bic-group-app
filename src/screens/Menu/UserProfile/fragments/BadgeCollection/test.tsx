import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
