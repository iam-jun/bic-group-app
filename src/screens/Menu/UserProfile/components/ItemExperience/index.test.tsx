import * as React from 'react';
import { renderWithRedux } from '~/test/testUtils';

import ItemExperience from '.';

describe('ItemExperience component', () => {
  const baseProps = {
    id: '1',
    company: 'company',
    titlePosition: 'titlePosition',
    startDate: '2022-03-07T07:58:05.436Z',
    currentlyWorkHere: true,
    endDate: null,
    location: 'location',
  };

  it('renders correctly', () => {
    const wrapper = renderWithRedux(<ItemExperience item={baseProps} isCurrentUser />);
    expect(wrapper).toMatchSnapshot();
  });
});
