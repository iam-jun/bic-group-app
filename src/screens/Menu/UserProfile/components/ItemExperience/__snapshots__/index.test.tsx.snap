// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemExperience component renders correctly 1`] = `
<View
  style={
    Object {
      "paddingBottom": 24,
    }
  }
  testID="item_experience"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingBottom": 16,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": undefined,
          "height": 22,
          "marginRight": 12,
          "width": 1,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 17,
          },
          Object {},
          undefined,
        ]
      }
    >
      Work at 
      <Text
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "BeVietnamPro-Medium",
              "fontSize": 17,
            },
            Object {},
            undefined,
          ]
        }
      >
        company
      </Text>
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 8,
        },
        undefined,
      ]
    }
    testID="info_item"
  >
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "BeVietnamPro-Light",
            "fontSize": 15,
          },
          Object {},
          undefined,
        ]
      }
      testID="info_item.title"
    >
      Title/Position
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 15,
          },
          Object {},
          undefined,
        ]
      }
      testID="info_item.value"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 8,
        },
        undefined,
      ]
    }
    testID="info_item"
  >
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "BeVietnamPro-Light",
            "fontSize": 15,
          },
          Object {},
          undefined,
        ]
      }
      testID="info_item.title"
    >
      Division
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 15,
          },
          Object {},
          undefined,
        ]
      }
      testID="info_item.value"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 8,
        },
        undefined,
      ]
    }
    testID="info_item"
  >
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "BeVietnamPro-Light",
            "fontSize": 15,
          },
          Object {},
          undefined,
        ]
      }
      testID="info_item.title"
    >
      Start Date
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 15,
          },
          Object {},
          undefined,
        ]
      }
      testID="info_item.value"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 8,
        },
        undefined,
      ]
    }
    testID="info_item"
  >
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "BeVietnamPro-Light",
            "fontSize": 15,
          },
          Object {},
          undefined,
        ]
      }
      testID="info_item.title"
    >
      End Date
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 15,
          },
          Object {},
          undefined,
        ]
      }
      testID="info_item.value"
    />
  </View>
</View>
`;
